src/Units/UnitModel.java

```diff
@@ -35,38 +35,62 @@
   public int maxFuel;
   public int idleFuelBurn;
   public int movePower;
+  public int vision;
+  public boolean piercingVision = false;
   public MoveType propulsion;
-  public ActionType[] possibleActions;
+  public ArrayList<ActionType> possibleActions = new ArrayList<ActionType>();
   public Set<TerrainType> healableHabs;
-  public WeaponModel[] weaponModels;
+  public ArrayList<WeaponModel> weaponModels = new ArrayList<WeaponModel>();
 
   public int maxHP;
   public int holdingCapacity;
   public Vector<UnitEnum> holdables;
   private int COstr;
   private int COdef;
 
-  public UnitModel(String pName, UnitEnum pType, ChassisEnum pChassis, int cost, int pFuelMax, int pIdleFuelBurn, int pMovePower, MoveType pPropulsion,
-      ActionType[] actions, WeaponModel[] weapons)
+  public UnitModel(String pName, UnitEnum pType, ChassisEnum pChassis, int cost, int pFuelMax, int pIdleFuelBurn, int pVision, int pMovePower,
+      MoveType pPropulsion, ActionType[] actions, WeaponModel[] weapons)
+  {
+    this(pName, pType, pChassis, cost, pFuelMax, pIdleFuelBurn, pVision, pMovePower, pPropulsion);
+
+    for( ActionType action : actions )
+    {
+      possibleActions.add(action);
+    }
+    for( WeaponModel action : weapons )
+    {
+      weaponModels.add(action);
+    }
+  }
+
+  public UnitModel(String pName, UnitEnum pType, ChassisEnum pChassis, int cost, int pFuelMax, int pIdleFuelBurn, int pVision, int pMovePower,
+      MoveType pPropulsion, ArrayList<ActionType> actions, ArrayList<WeaponModel> weapons)
```

Comment:
Is this constructor actually used anywhere? I guess I could check this branch out and look, but

Reply:
It's used by the cloning logic